File: examples/sync/kraken_public.py
import dcex


def main() -> None:
    client = dcex.kraken()

    server_time = client.get_server_time()
    print(server_time)

    orderbook = client.get_spot_orderbook(product_symbol="BTC-USDT-SPOT", count=5)
    print(orderbook)


if __name__ == "__main__":
    main()

File: crates/dcex/examples/kraken_public.rs
use std::time::Duration;

use dcex::exchanges::kraken::KrakenClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = KrakenClient::public(Duration::from_secs(10))?;
    let response = client
        .get_spot_ticker()
        .param("product_symbol", "BTC-USDT-SPOT")
        .await?;
    println!("{}", response.data);
    Ok(())
}
